src/Featurizers/UnitTests/TruncatedSVDFeaturizer_UnitTests.cpp

```diff
@@ -0,0 +1,192 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#include "../../Featurizers/TruncatedSVDFeaturizer.h"
+#include "../TestHelpers.h"
+#include "../../Archive.h"
+
+namespace NS = Microsoft::Featurizer;
+
+template<typename InputT, typename TransformedT>
+void EstimatorTest(InputT const &trainingMatrix,
+                   TransformedT const &singularValuesLabel,
+                   TransformedT const &singularVectorsLabel,
+                   std::float_t eps = 0.0001f) {
+
+    using SVDComponentsEstimator                       = NS::Featurizers::SVDComponentsEstimator<InputT, TransformedT>;
+    using SVDComponentsAnnotationData                  = NS::Featurizers::SVDComponentsAnnotationData<TransformedT>;
+
+    NS::AnnotationMapsPtr const                        pAllColumnAnnotations(NS::CreateTestAnnotationMapsPtr(1));
+    SVDComponentsEstimator                             estimator(pAllColumnAnnotations, 0);
+
+    estimator.begin_training();
```

Comment:
>estimator.begin_training(); [](start = 4, length = 27)

Is it possible to use TestHelper method or its specialization Train<>()?

Reply:
For now it's not, but I can extend that function

Reply:
I agree with Dmitri - we should use or extend Train rather than inventing a one-off solution.